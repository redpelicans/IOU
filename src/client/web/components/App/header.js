import React from 'react';
import styled from 'styled-components';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import AccountCircle from '@material-ui/icons/AccountCircle';
import History from '@material-ui/icons/History';

const StyledTypography = styled(Typography)`
  flex: 1;
`;

const Header = () => {
  return (
    <AppBar position="static">
      <Toolbar>
        <StyledTypography color="inherit" variant="title">
          IOU by RedPelicans
        </StyledTypography>
        <IconButton color="inherit">
          <History />
        </IconButton>
        <IconButton color="inherit">
          <AccountCircle />
        </IconButton>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
