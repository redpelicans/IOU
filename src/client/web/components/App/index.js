import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { map, values } from 'ramda';
import routes, { defaultRoute } from '../../routes';
import Header from './header';
import Events from '../../pages/Events';

const App = () => {
  return (
    <div>
      <Header />
      <div>
        <Switch>
          {map(
            ({ exact, path, component }) => (
              <Route
                key={path}
                exact={exact}
                path={path}
                component={component}
              />
            ),
            values(routes),
          )}
          <Route path="/" component={defaultRoute.component} />
        </Switch>
      </div>
    </div>
  );
};

export default App;
