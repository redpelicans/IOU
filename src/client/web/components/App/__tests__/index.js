import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';

import App from '../index';
import reducer from '../../../reducers';

describe('App', () => {
  const store = createStore(reducer);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>,
      div,
    );
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
