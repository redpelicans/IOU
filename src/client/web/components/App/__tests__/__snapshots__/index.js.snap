// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <Jss(Header) />
  <div>
    <Switch>
      <Route
        exact={true}
        key="/about"
        path="/about"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/events"
        path="/events"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/event/:id"
        path="/event/:id"
      />
      <Route
        component={[Function]}
        path="/"
      />
    </Switch>
  </div>
</div>
`;
